0011/README.md

```diff
@@ -0,0 +1,62 @@
+---
+id: 0011
+title: how to announce bc break for Tinkerbell
+status: discussion
+authors: Gianluca Arbezzano <gianarb92@gmail.com>
+---
+
+## Summary
+
+At this stage, where the various services are not released and do not have
+semver we can accept BC break.
+
+We need to have a consolidate way to communicate them.
+
+## Goals and no-Goals
+
+Goal:
+
+* Clarify the way we can communicate bc break across services and in Tinkerbell.
+* A solution with low overhead that has to work NOW.
+
+No-Goal:
+
+* Explain how SemVer manage Bc-Break or how we will communicate bc-break when
+  SemVer will be in use.
+* Explain how a component should communicate bc-break to their community.
+
+## Content
+
+Even if we do not expect many BC break the possibility to remove or change a
+functionality is a luxury that won't stay for long.
+
+Even if we don't have a day for the first major release of Tinkerbell or for the
+other component I doubt it will be next week, this means that BC break are
+expected and we need to communicate them effectively.
+
+Every PR that introduce a BC break will get labeled with a `bc-break` label. In
+this way we will be able to build a workflow that programmatically will extract
+the BC-Break for every component.
+
+Every PR has to contain as part of the description a section titled:
+"how to migrate" teaching the user about how it should pass from a previous
+version to a new one when possible. If not possible because the section will
+explain why we remove the entire functionality and will offer an alternative if
+it exists.
```

Comment:
I think even better would be to have a code snippet be a soft-requirement, similar to how `go fix` used to exist to handle this before Go1. I don't think it needs to be a specific tool, or expected to always be 100% correct dealing with odd OSes (non-gnu sed for example). It would a serve the purpose of "how to mitigate" by showing with code that can either be run or "ported".  Nothing too onerous either, sed, awk, bash, python (maybe perl), even go code itself maybe.

Maybe something like https://en.wikipedia.org/wiki/Coccinelle_(software) or similar would be a good option. I feel like there's something along the lines of coccinelle but easier/friendlier language maybe, but I can't seem to find anything.

Reply:
Yep! That's nice. I don't want to go too deep on "How the migrate section should be written" mainly because I can't really guess all the possible bc-break. Maybe the bc is an HTTP API field, maybe a code one, maybe a deprecated feature, so it won't be possible to migrate.

Who knows! It looks like a deep rabbit hole to follow right now. But I think reviewers and lesson learned plays a good lesson here. WE have to be sure that the migrate chapter is well written, what ever it means

Reply:
Yeah those are good points. Somethings can't be easily specified in code. But I think having a call out to add a script or similar when it *is* possible would make things easier. For example https://github.com/tinkerbell/tink/pull/280 can have a sed script adding a default timeout.